### PR TITLE
Modified execution result in case of exceptions in C/C++ executables.

### PR DESCRIPTION
**Pull request by Innos**
Closes https://github.com/SoftUni-Internal/suls-issues/issues/4218 - Fix problem with stack overflow exceptions in C/C++

### CODE REVIEW

File: Open Judge System/Workers/OJS.Workers.Executors/RestrictedProcessExecutor.cs [hunk 1]
```diff
@@ -167,6 +167,12 @@ public RestrictedProcessExecutor(int baseTimeUsed, int baseMemoryUsed)
 
                 // Report exit code and total process working time
                 result.ExitCode = restrictedProcess.ExitCode;
+
+                if (result.ExitCode != 0 && result.ExitCode != -1)
```
Comment (gogo4ds) [inline Open Judge System/Workers/OJS.Workers.Executors/RestrictedProcessExecutor.cs]: Why not include this check in the `if` statement that checks if the `result.ErrorOutput` `IsNullOrEmpty` and also makes the type `RunTimeError` and maybe make it the last check in the executor, to be sure that no other result will override the `RunTimeError` output

**Comment by Innos** [comment]
Added the same exit code check to the Standard Process Executor, so I ask for a new review.

File: Open Judge System/Workers/OJS.Workers.Executors/RestrictedProcessExecutor.cs [hunk 2]
```diff
@@ -191,6 +191,13 @@ public RestrictedProcessExecutor(int baseTimeUsed, int baseMemoryUsed)
             {
                 result.Type = ProcessExecutionResultType.RunTimeError;
             }
+            else
+            {
+                if (result.ExitCode != 0 && result.ExitCode != -1)
```
Comment (gogo4ds) [inline Open Judge System/Workers/OJS.Workers.Executors/RestrictedProcessExecutor.cs]: just add this check litteraly in the above `if(..)` with `||` no need of `else` statement

File: Open Judge System/Workers/OJS.Workers.Executors/StandardProcessExecutor.cs [hunk 3]
```diff
@@ -211,6 +211,13 @@ public StandardProcessExecutor(int baseTimeUsed, int baseMemoryUsed)
             {
                 result.Type = ProcessExecutionResultType.RunTimeError;
             }
+            else
+            {
+                if (result.ExitCode != 0 && result.ExitCode != -1)
```
Comment (gogo4ds) [inline Open Judge System/Workers/OJS.Workers.Executors/StandardProcessExecutor.cs]: just add this check litteraly in the above `if(..)` with `||` no need of `else` statement

**Pull request merged**